source/browseMode.py

```diff
@@ -2001,6 +2041,129 @@ def _iterNotLinkBlock(self, direction="next", pos=None):
 				yield TextInfoQuickNavItem("notLinkBlock", self, textRange)
 			item1=item2
 
+	STYLE_ATTRIBUTES = frozenset([
+		'background-color',
+		'color',
+		'font-family',
+		'font-size',
+		'bold',
+		'italic',
+		'marked',
+		'strikethrough',
+		'text-line-through-style',
+		'underline',
+		'text-underline-style',
+	])
+
+	def _extractStyles(
+			self,
+			info: textInfos.TextInfo,
+	) -> "textInfos.TextInfo.TextWithFieldsT":
+		"""
+		This function calls TextInfo.getTextWithFields(), and then processes fields in the following way:
+		1. Highlighted(marked) text is currently reported as Role.MARKED_CONTENT, and not formatChange.
+		For ease of further handling we create a new boolean format field "marked"
+		and set its value according to presence of Role.MARKED_CONTENT.
+		2. Then we drop all control fields, leaving only formatChange fields and text.
+		"""
+		stack = [{}]
+		result = []
+		for field in info.getTextWithFields():
+			if isinstance(field, textInfos.FieldCommand):
+				if field.command == 'controlStart':
+					style = {**stack[-1]}
+					if field.field.get('role') == controlTypes.Role.MARKED_CONTENT:
+						style['marked'] = True
+					stack.append(style)
+				elif field.command == 'controlEnd':
+					del stack[-1]
+				elif field.command == 'formatChange':
+					field.field = {
+						k: v
+						for k, v in {**field.field, **stack[-1]}.items()
+						if k in self.STYLE_ATTRIBUTES
+					}
+					result.append(field)
+				else:
+					raise RuntimeError()
+			elif isinstance(field, str):
+				result.append(field)
+			else:
+				raise RuntimeError

```

Comment:
can you note that this raises a runtime error in the docstring?
Please add a relevant message to the runtime error